tests/unit/s2n_server_psk_extension_test.c

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+
+#include "crypto/s2n_hmac.h"
+#include "tls/extensions/s2n_server_psk.h"
+
+/* Include source to test static methods. */
+#include "tls/extensions/s2n_server_psk.c"
+
+struct s2n_psk_test_case {
+    s2n_hmac_algorithm hmac_alg;
+    uint8_t hash_size;
+    const uint8_t* identity;
+    size_t identity_size;
+    const uint8_t* secret;
+    size_t secret_size;
+
+};
+
+uint8_t test_identity_1[] = "test identity";
+uint8_t test_identity_2[] = "another identity";
+uint8_t test_secret_1[] = "test secret";
+uint8_t test_secret_2[] = "another secret";
+
+struct s2n_psk_test_case test_cases[] = {
+    { .hmac_alg = S2N_HMAC_SHA384, .hash_size = SHA384_DIGEST_LENGTH,
+      .identity = test_identity_1, .identity_size = sizeof(test_identity_1),
+      .secret = test_secret_1, .secret_size = sizeof(test_secret_1)},
+    { .hmac_alg = S2N_HMAC_SHA256, .hash_size = SHA256_DIGEST_LENGTH,
+      .identity = test_identity_2, .identity_size = sizeof(test_identity_2),
+      .secret = test_secret_2, .secret_size = sizeof(test_secret_2)},  
+};
+
+static s2n_result validate_psk_is_wiped(struct s2n_psk *psk)
+{
+    ENSURE_EQ(psk->identity.data, NULL);
+    ENSURE_EQ(psk->identity.size, 0);
+    ENSURE_EQ(psk->secret.data, NULL);
+    ENSURE_EQ(psk->secret.size, 0);
+    ENSURE_EQ(psk->early_secret.data, NULL);
+    ENSURE_EQ(psk->early_secret.size, 0);
+
+    return S2N_RESULT_OK;
+}
+
+static s2n_result validate_psk_is_not_wiped(struct s2n_psk *psk)
+{
+    ENSURE_REF(psk->identity.data);
+    ENSURE_NE(psk->identity.size, 0);
+    ENSURE_REF(psk->secret.data);
+    ENSURE_NE(psk->secret.size, 0);
+
+    return S2N_RESULT_OK;
+}
+
+static s2n_result setup_psks(struct s2n_connection *conn)
```

Comment:
In this method we only setup PSKs for one connection - either client/server, used within the  `s2n_server_psk_send`/`s2n_server_psk_recv` unit tests.  In `setup_order_mismatch_psks` we setup PSKs for both the client and server with the order of PSKs in the psk list mismatched.

Reply:
Then can you combine the two setup methods? `setup_order_mismatch_psks` can call `setup_psks` to handle setting up one of the connections. I'd also suggest either removing test_psks completely or using the same model for `setup_order_mismatch_psks`.

Right now you have two completely separate methods of performing very similar setup. I think this can be simplified.